### ISSUE TITLE
In lesson 3 Perceptron.ipynb, there is no learning rate in the train function

### ISSUE THREAD

**Issue by tsreaper** [microsoft/AI-For-Beginners#242]
As mentioned in `Perceptron.ipynb`, the gradient descent algorithm comes with a learning rate. However there is no learning rate in the `train` function.

**Comment by cutePanda123** [comment]
i could be wrong but i feel the learning rate is 1 this is why it is ignored.

**Comment by leestott** [comment]
You're absolutely right, while the notebook mentions gradient descent and its learning rate, the `train` function in the code does

###  Gradient Descent and Learning Rate

In theory, **gradient descent** updates weights using the formula:

```
weight = weight - learning_rate * gradient
```

The **learning rate** controls how big each update step is. A smaller rate means slower learning but more stability; a larger rate speeds up learning but risks overshooting.

### What's Missing in the Notebook

In the `Perceptron.ipynb` notebook, the `train` function updates weights like this:

```python
self.weights += self.learning_rate * error * x
```

But in the actual code provided, **`self.learning_rate` is never defined**, nor is a learning rate passed into the `train` function or the `Perceptron` class constructor. This means:

- The code either assumes a default learning rate (which is not shown),
- Or it omits the learning rate entirely, making the update effectively:
  ```python
  self.weights += error * x
  ```
  —which is equivalent to a learning rate of 1.

### What You Can Do

To and align with the theory:

1. **Add a learning rate parameter** to the `Perceptron` class:
   ```python
   class Perceptron:
       def __init__(self, input_size, learning_rate=0.01):
           self.weights = np.zeros(input_size)
           self.learning_rate = learning_rate
   ```

2. **Use it in the weight update**:
   ```python
   self.weights += self.learning_rate * error * x
   ```

This would make the implementation consistent with the gradient descent algorithm.
